internal/common/utils/sets.go

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2019 Palantir Technologies. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+// StringSet is a non-thread safe set of unique strings
+type StringSet struct {
```

Comment:
It might be a bit clearer if you define this as `type StringSet map[string]bool`, that way you can get rid of the struct and don't have a pointer to a map